### PR TITLE
[Network] az network virtual-appliance: Support Azure network virtual appliance.

### LINKED ISSUE #13947: AZ CLI support for Network Virtual Appliance
**Resource Provider**
NFVRP (Microsoft.Network)

**Description of Feature or Work Requested**
NetworkVirtualAppliance is a new service that we want to onboard. Each NetworkVirtualAppliance has child resources called virtualApplianceSites and also need to support GetNetworkVirtualApplianceSkus

**Minimum API Version Required**
2020-05-01

**Swagger Link**
https://github.com/Azure/azure-rest-api-specs/pull/9682
https://github.com/Azure/azure-rest-api-specs/pull/8270
https://github.com/Azure/azure-rest-api-specs/pull/8010
https://github.com/Azure/azure-rest-api-specs/pull/9357

**Target Date**
Release Target date 07/15/2020

### PR DESCRIPTION
**Pull request by MyronFanQiu**
**Description**  

Fix #13947 

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli/azure/cli/command_modules/network/commands.py [hunk 1]
```diff
@@ -1220,3 +1240,23 @@ def _make_singular(value):
...
+        g.generic_update_command('update', custom_func_name='update_network_virtual_appliance')
+        g.show_command('show', 'get')
+        g.custom_command('list', 'list_network_virtual_appliance')
+        g.command('delete', 'delete', confirmation=True)
+
+    with self.command_group('network virtual-appliance site', virtual_appliance_sites_sdk, is_preview=True) as g:
+        g.custom_command('create', 'create_network_virtual_appliance_site')
+        g.generic_update_command('update', custom_func_name='update_network_virtual_appliance_site')
+        g.show_command('show', 'get')
+        g.command('delete', 'delete')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/commands.py:1256]: confirm for delete?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -5933,3 +5933,80 @@ def list_security_partner_provider(cmd, resource_group_name=None):
...
+
+
+# region network virtual appliance
+def create_network_virtual_appliance(cmd, resource_group_name, network_virtual_appliance_name,
+                                     vendor, bundled_scale_unit, market_place_version,
+                                     virtual_hub, boot_strap_configuration_blobs=None,
+                                     cloud_init_configuration_blobs=None,
+                                     cloud_init_configuration=None, asn=None,
+                                     location=None, tags=None, no_wait=False):
+    client = network_client_factory(cmd.cli_ctx).network_virtual_appliances
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:5945]: define "client" as method argument

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 3]
```diff
@@ -768,7 +768,7 @@ def load_arguments(self, _):
 
     with self.argument_context('network express-route port identity assign', arg_group='Identity', min_api='2019-08-01') as c:
         c.argument('user_assigned_identity', options_list='--identity',
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:770]: define it as CLIArgumentType for reuse?

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 4]
```diff
@@ -1844,3 +1845,28 @@ def load_arguments(self, _):
             c.argument('resource_group_name', required=False)
             c.argument('resource_name', required=False, help='Name of the resource')
     # endregion
+
+    # region Network Virtual Appliance
+    with self.argument_context('network virtual-appliance') as c:
+        c.argument('network_virtual_appliance_name', help='The name of Network Virtual Appliance', options_list=['--name', '-n'])
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:1851]: define it as CLIArgumentType for reuse?

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 5]
```diff
@@ -6113,3 +6113,94 @@
...
+short-summary: Manage Azure Network Virtual Appliance.
+"""
+
+helps['network virtual-appliance create'] = """
+type: command
+short-summary: Create an Azure network virtual appliance.
+examples:
+  - name: Create an Azure network virtual appliance.
+    text: |
+        az network virtual-appliance create -n MyName -g MyRG --vhub {vhubID} --vendor "barracudasdwanrelease" --scale-unit 2 -v latest --asn 10000 --init-config "echo $hello" --boot-blobs {blobUrl} {blobUrl} --cloud-blobs {blobUrl} {blobUrl}
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:6128]: use different urls

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 6]
```diff
@@ -1844,3 +1845,28 @@ def load_arguments(self, _):
...
+        c.argument('vendor', help='Virtual Appliance Vendor.')
+        c.argument('bundled_scale_unit', options_list=['--scale-unit'], help='Virtual Appliance Scale Unit.')
+        c.argument('market_place_version', options_list=['--version', '-v'], help='Virtual Appliance Version.')
+        c.argument('boot_strap_configuration_blobs', options_list=['--boot-strap-config-blobs', '--boot-blobs'], nargs='+', help='Space-separated list of BootStrapConfigurationBlobs storage URLs.')
+        c.argument('cloud_init_configuration_blobs', options_list=['--cloud-init-config-blobs', '--cloud-blobs'], nargs='+', help='Space-separated list of CloudInitConfigurationBlob storage URLs.')
+        c.argument('virtual_hub', options_list=['--vhub'], help='Name or ID of the virtual hub to which the Security Partner Provider belongs.', validator=validate_virtual_hub)
+        c.argument('cloud_init_configuration', options_list=['--cloud-init-config', '--init-config'], help='CloudInitConfiguration string in plain text.')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:1858]: help: The scripts that will be run during cloud initialization?

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 7]
```diff
@@ -1844,3 +1845,28 @@ def load_arguments(self, _):
             c.argument('resource_group_name', required=False)
             c.argument('resource_name', required=False, help='Name of the resource')
     # endregion
+
+    # region Network Virtual Appliance
+    with self.argument_context('network virtual-appliance') as c:
+        c.argument('network_virtual_appliance_name', help='The name of Network Virtual Appliance', options_list=['--name', '-n'])
+        c.argument('vendor', help='Virtual Appliance Vendor.')
+        c.argument('bundled_scale_unit', options_list=['--scale-unit'], help='Virtual Appliance Scale Unit.')
+        c.argument('market_place_version', options_list=['--version', '-v'], help='Virtual Appliance Version.')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:1854]: sku argument group

File: src/azure-cli/azure/cli/command_modules/network/_params.py (hunk 3 above)
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/_params.py:770]: will do it in later PR.

**Pull request merged**